opening_hours/utils/hauki_api_client.py

```diff
@@ -1,31 +1,127 @@
+import datetime
 import json
-from typing import Literal
-from urllib.parse import urljoin
+from typing import TypedDict
 
 from django.conf import settings
 from requests import Response, request
 
 from opening_hours.errors import HaukiAPIError, HaukiConfigurationError, HaukiRequestError
+from opening_hours.utils.hauki_api_types import (
+    HaukiAPIOpeningHoursResponse,
+    HaukiAPIOpeningHoursResponseItem,
+    HaukiAPIResource,
+    HaukiAPIResourceListResponse,
+)
 from tilavarauspalvelu.utils import logging
 
+__all__ = [
+    "HaukiAPIClient",
+]
+
 REQUESTS_TIMEOUT = 15
 
 logger = logging.getLogger(__name__)
 
 
+class HaukiGetResourcesParams(TypedDict):
+    resource_ids: str  # Comma separated list of UUIDs. Supports [data_source_id]:[origin_id] style ids
+    parent: str  # ID of the parent resource
+    child: str  # ID of a child resource
+    data_source: str  # ID of the data source
+    origin_id_exists: bool  # Filter by existing/missing origin_id
+    ordering: str
+    page: int
+    page_size: int
+
+
 class HaukiAPIClient:
-    """Hauki API client, minimal functionality implemented."""
+    ############
+    # resource #
+    ############
+
+    @classmethod
+    def get_resources(
+        cls,
+        *,
+        hauki_resource_ids: list[int],
+        **kwargs: HaukiGetResourcesParams,
```

Comment:
This is incorrect, `kwargs` are now typed as `dict[str, HaukiGetResourcesParams]`. This is the reason python 3.12 added [typing.Unpack](https://docs.python.org/3/library/typing.html#typing.Unpack). For now, you need to  untype the kwags, and add the annotation in the method body like I showed (or add [typing_extensions](https://pypi.org/project/typing-extensions/) and use it that way).